src/app/wallets/update-on-chain-receipt.ts

```diff
@@ -6,14 +6,71 @@ import { NotificationsService } from "@services/notifications"
 import { LedgerService } from "@services/ledger"
 import { OnChainError, TxFilter, TxDecoder } from "@domain/bitcoin/onchain"
 import { toLiabilitiesAccountId } from "@domain/ledger"
-
 import { redlock } from "@core/lock"
+import { ONCHAIN_LOOK_BACK, ONCHAIN_MIN_CONFIRMATIONS, BTC_NETWORK } from "@config/app"
+
+export const updateOnChainReceipt = async ({
+  scanDepth = ONCHAIN_LOOK_BACK,
+  logger,
+}: {
+  scanDepth?: number
+  logger: Logger
+}): Promise<number | ApplicationError> => {
+  const onChain = OnChainService(TxDecoder(BTC_NETWORK))
+  if (onChain instanceof OnChainError) {
+    return onChain
+  }
+
+  const onChainTxs = await onChain.getIncomingTransactions(scanDepth)
+  if (onChainTxs instanceof OnChainError) {
+    return onChainTxs
+  }
+
+  const walletRepo = WalletsRepository()
+  const logError = ({ walletId, txId, error }) => {
+    logger.error(
+      { walletId, txId, error },
+      "Could not updateOnChainReceipt from updateOnChainReceiptForWallet",
+    )
+  }
+
+  let wallets, addresses, walletId, txId, result
+  for (const tx of onChainTxs) {
+    txId = tx.rawTx.id
+
+    addresses = tx.rawTx.outs.reduce<OnChainAddress[]>((a, o) => {
+      if (o.address) a.push(o.address)
+      return a
+    }, [])
+
+    wallets = await walletRepo.findByAddresses(addresses)
```

Comment:
does not need cursor because the amount of wallets updated in a transaction maybe is less than the first batch in a cursor query